File: L2S3/PWI/TP4/src/Personne.php
<?php

class Personne{

	private int $id;
	private string $nom;
	private string $prenom;

	public function __construct(array $data){
		$this->hydrate($data);
	}

	// GETTERS
	public function getID(): int{ return $this->id; }
	public function getNom(): string{ return $this->nom; }
	public function getPrenom(): string{ return $this->prenom; }

	// SETTERS
	public function setNom(string $nom): void{ $this->nom = $nom; }
	public function setPrenom(string $prenom): void{ $this->prenom = $prenom; }

	public function hydrate(array $data){
		$this->id = intval($data["id"]);
		$this->setNom($data["nom"]);
		$this->setPrenom($data["prenom"]);
	}

	// FUNCTIONS
	public function toArray(): array{
		return [
			"id" => $this->id,
			"nom" => $this->nom,
			"prenom" => $this->prenom
		];
	}
	public function __toString(): string{
		return sprintf("%s %s est d'identifiant %d.",
			$this->prenom, $this->nom, $this->id
		);
	}
}

File: L1S2/POO/TD4/src/Personne.java
public class Personne {

	private int numero;
	private String nom;
	private String prenom;
	private int age;
	
	public Personne(int numero, String nom, String prenom, int age){
		this.numero = numero;
		this.nom = nom;
		this.prenom = prenom;
		this.age = age;
	}

	// Getters
	public int getNumero(){ return numero; }
	public String getNom(){ return nom; }
	public String getPrenom(){ return prenom; }
	public int getAge(){ return age; }

	// Functions
	@Override
	public String toString() {
		return getNom() + " " + getPrenom();
	}
}
